src/tools/immutable_object_cache/Types.h

```diff
@@ -0,0 +1,60 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_CACHE_TYPES_H
+#define CEPH_CACHE_TYPES_H
+
+#include "include/encoding.h"
+#include "include/Context.h"
+
+namespace ceph {
+namespace immutable_obj_cache {
+
+struct ObjectCacheMsgHeader {
+    uint64_t seq;                         /* sequence id */
+    uint16_t type;                        /* msg type */
+    uint16_t version;                     /* object cache version */
+    uint32_t padding;
+    uint32_t data_len;
+    uint32_t reserved;
+
+    void encode(bufferlist& bl) const;
+    void decode(bufferlist::const_iterator& it);
+};
+
+class ObjectCacheMsgData {
+public:
+  uint64_t m_image_size;
+  uint64_t m_read_offset;
+  uint64_t m_read_len;
+  std::string m_pool_name;
```

Comment:
Replace with ```int64_t``` pool id (names can be changed) -- and include a ```std::string pool_namespace``` locator.

Reply:
Thanks, do we need to include snap_id here also? Is it safe to index one cache file with `pool_id:snap_id:pool_namspace:oid`?

Reply:
Good point. Yes, it should also include the snap id.

Reply:
fix pushed